core/Config.php

```diff
@@ -455,4 +454,24 @@ public static function setSetting($sectionName, $name, $value)
         $section[$name] = $value;
         self::getInstance()->$sectionName = $section;
     }
-}
+
+    /**
+     * Sanity check a config file by checking its size and content
+     *
+     * @param string $localPath
+     * @return bool
+     */
+    public function sanityCheck(string $localPath): bool
+    {
+        $content = @file_get_contents($localPath);
+        $filesize = filesize($localPath);
+
+        if ($filesize < 1500 || strpos($content, '[database]') === false) {
```

Comment:
fyi I'm thinking we might actually only want to check the `[database]` if also `$output` above contained this section in the string to write.

Reason being that I just noticed eg in Matomo for WordPress we don't save a `[database]` section as it reads the WordPress DB credentials. Also some might actually define the DB using this plugin so there wouldn't be a DB section https://github.com/matomo-org/plugin-EnvironmentVariables

Maybe the sanity check would purely need to check if `trim($output)=== trim($content)`?